tests/api/test_aws.py

```diff
@@ -1,144 +1,471 @@
 import json
-from unittest.case import TestCase
-from unittest.mock import MagicMock, call, patch
+import os
+from unittest.mock import MagicMock
 
-from botocore.exceptions import ClientError
+import boto3
+from django.conf import settings
+from model_mommy import mommy
+import moto
+import pytest
 
-from controlpanel.api.aws import aws
+from controlpanel.api import aws
 
 
-@patch.object(aws, 'client', MagicMock())
-class AwsTestCase(TestCase):
+@pytest.yield_fixture(autouse=True)
+def enable_db_for_all_tests(db):
+    pass
 
-    def test_create_bucket(self):
-        aws.create_bucket('bucketname', 'eu-west-1')
-        aws.client.return_value.create_bucket.assert_called()
 
-    def test_put_bucket_logging(self):
-        aws.put_bucket_logging('bucketname', 'target_bucket', 'target_prefix')
-        aws.client.return_value.put_bucket_logging.assert_called()
+@pytest.fixture(autouse=True)
+def aws_creds():
+    os.environ['AWS_ACCESS_KEY_ID'] = 'testing'
+    os.environ['AWS_SECRET_ACCESS_KEY'] = 'testing'
+    os.environ['AWS_SECURITY_TOKEN'] = 'testing'
+    os.environ['AWS_SESSION_TOKEN'] = 'testing'
 
-    def test_get_inline_policy_document_when_found(self):
-        role_name = 'test-user-rolename'
-        policy_name = 's3-access'
-        policy_document = {'test': 'policy document'}
 
-        aws.client.return_value.get_role_policy.return_value = {
-            'PolicyDocument': policy_document,
-        }
+@pytest.yield_fixture(autouse=True)
+def iam(aws_creds):
+    with moto.mock_iam():
+        yield boto3.resource('iam')
```

Comment:
how does `autouse=True` works in this case? Probably a stupid question, would this mock IAM within this file right? not across all the tests right?

Reply:
`autouse=True` means the fixture is evaluated for every test function in this file. If this file was `conftest.py`, then the fixture would be evaluated for every test function in this directory and subdirectories.